lib/prx_auth/certificate_cache.ex

```diff
@@ -0,0 +1,47 @@
+defmodule PrxAuth.Certificate.Cache do
```

Comment:
this seems more generic than to only be used for certs, perhaps reflect that in the namespace?